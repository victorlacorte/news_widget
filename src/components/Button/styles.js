import { memo } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const StyledButton = styled.button.attrs((props) => ({
  // type: props.type || 'button',
  disabled: props.disabled || false,
  color: props.color || 'blue',
}))`
  box-sizing: border-box;
  min-height: 40px;
  padding: 0px 32px;
  border: none;
  border-radius: 4px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  outline: none;
  color: white;

  ${({ theme }) => theme.breakpoints.down('sm')`
    width: 100%;
  `}

  ${({ theme }) => css`
    ${theme.typography.button};
  `}

  ${({ disabled, color }) => !disabled && css`
    background-color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].dark};
    }
  `}

  ${({ disabled }) => disabled && css`
    pointer-events: none;
    cursor: default;
    background-color: ${({ theme }) => theme.palette.grey.dark};
  `}
`;

StyledButton.propTypes = {
  // type: PropTypes.oneOf(['button', 'submit']),

  disabled: PropTypes.bool,

  color: PropTypes.oneOf(['blue', 'black', 'grey']),
};

export default memo(StyledButton);
