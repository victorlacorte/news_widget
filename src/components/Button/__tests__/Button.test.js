import React from 'react';
import { shallow } from 'enzyme';

import Button from '..';

describe('<Button />', () => {
  it('creates a Button component', () => {
    const wrapper = shallow(<Button />);

    expect(wrapper).toMatchSnapshot();
  });

  it('creates a disabled Button component', () => {
    const wrapper = shallow(<Button disabled />);

    expect(wrapper.props().disabled).toBe(true);
  });
});
