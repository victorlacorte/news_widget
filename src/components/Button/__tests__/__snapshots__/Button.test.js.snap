// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> creates a Button component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  box-sizing: border-box;
  min-height: 40px;
  padding: 0px 32px;
  border: none;
  border-radius: 4px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  outline: none;
  color: white;

  ",
          [Function],
          "

  ",
          [Function],
          "

  ",
          [Function],
          "

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "color": [Function],
        "disabled": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
